src/map/src/cmapclient.cpp

```diff
@@ -191,6 +194,35 @@ bool CMapClient::ChatReply(std::unique_ptr<RoseCommon::CliChat> P) {
   return true;
 }
 
+bool CMapClient::ChatWhisper(std::unique_ptr<RoseCommon::CliWhisper> P) {
+  logger_->trace("CMapClient::ChatWhisper()");
+
+  if (login_state_ != eSTATE::LOGGEDIN) {
+    logger_->warn("Client {} is attempting to whisper before logging in.",
+                  GetId());
+    return true;
+  }
+
+  uint16_t _charID = GetId();
+  std::string _sender_name = entity_.component<BasicInfo>()->name_; 
+	std::string _recipient_name = P->targetId();
+	_recipient_name.erase(_recipient_name.size()-1); // for some reason this string has an extra char
+	std::string _message = P->message();
+	char *_recipient_client = NULL;
+	entitySystem_->process<BasicInfo, SocketConnector>([this, &_sender_name, &_recipient_client, _recipient_name] (Entity entity) {
+		if(_recipient_name == entity.component<BasicInfo>()->name_)
+			_recipient_client = (char *)entity.component<SocketConnector>()->client_;
+		//if(entity.component<SocketConnector>()->client_ == this)
+		//	_sender_name = entity.component<BasicInfo>()->name_;
+	});
+  logger_->trace("{} ({}) is whispering '{}' to {}", _sender_name, _charID,  _message, _recipient_name);
+	if(_recipient_client) {
+		auto packet = makePacket<ePacketType::PAKWC_WHISPER_CHAT>(_message, _sender_name);
+  	CMapServer::SendPacket((CMapClient *)_recipient_client, CMapServer::eSendType::PLAYER, *packet);
```

Comment:
you can call directly _recipient_client->Send() instead. It's easier to understand I think.